### DISCUSSION TITLE
How do you access flag values in custom commands?

### DISCUSSION THREAD

**Discussion by TravisCarden** [ddev/ddev#6430] [Q&A] [answered]
I have the following custom command definition. How do I access the value of the `spec` flag from within the script?

```shell
## Description: Run Experience Builder's Cypress tests.
## Usage: xb-cy-run
## Flags: [{"Name":"spec","Shorthand":"s","Usage":"run specific spec (files)","Type":"string","DefValue":"*/*.cy.js"}]
```

**Comment by rfay** [comment, marked as answer]
Hi @TravisCarden , welcome. If you look at the [launch](https://github.com/ddev/ddev/blob/16d38e33a27a9205903956b70c9be709e3f4d572/pkg/ddevapp/global_dotddev_assets/commands/host/launch#L42-L71) command, you'll see that it's ordinary bash argument handling. You can read about that in most bash tutorials, but launch is probably good enough to get you going. 

Mostly the `Flags` directive is just there to provide hints to `ddev help <command>` about what it should say.

**Comment by TravisCarden** [comment]
Thanks, @rfay! I thought that might be the case, but I hated to assume and embarrass myself. 😉
